datagateway_api/src/search_api/helpers.py

```diff
@@ -14,44 +16,154 @@
 
 
 @client_manager
-def get_search(endpoint_name, entity_name, filters):
+def get_search(entity_name, filters):
+    """
+    Search for data on the given entity, using filters from the request to restrict the
+    query
+
+    :param entity_name: Name of the entity requested to query against
+    :type entity_name: :class:`str`
+    :param filters: The list of Search API filters to be applied to the request/query
+    :type filters: List of specific implementation :class:`QueryFilter`
+    :return: List of records (in JSON serialisable format) of the given entity for the
+        query constructed from that and the request's filters
+    """
+
+    log.info("Searching for %s using request's filters", entity_name)
     log.debug("Entity Name: %s, Filters: %s", entity_name, filters)
 
-    icat_relations = mappings.get_icat_relations_for_panosc_non_related_fields(
-        entity_name,
-    )
-    # Remove any duplicate ICAT relations
-    icat_relations = list(dict.fromkeys(icat_relations))
-    if icat_relations:
-        filters.append(PythonICATIncludeFilter(icat_relations))
-
     query = SearchAPIQuery(entity_name)
 
     filter_handler = FilterOrderHandler()
     filter_handler.add_filters(filters)
+    icat_relations = filter_handler.add_icat_relations_for_panosc_non_related_fields(
+        entity_name,
+    )
+    filter_handler.add_icat_relations_for_non_related_fields_of_panosc_related_entities(
+        entity_name,
+    )
+    filter_handler.merge_python_icat_limit_skip_filters()
     filter_handler.apply_filters(query)
 
-    log.debug("Python ICAT Query: %s", query.icat_query.query)
-    # TODO - `getApiVersion()` used as a placeholder for testing client handling
-    # Replace with endpoint functionality when implementing the endpoints
-    return SessionHandler.client.getApiVersion()
+    log.debug("JPQL Query to be sent/executed in ICAT: %s", query.icat_query.query)
+    icat_query_data = query.icat_query.execute_query(SessionHandler.client, True)
+
+    panosc_data = []
+    for icat_data in icat_query_data:
+        panosc_model = getattr(models, entity_name)
+        panosc_record = panosc_model.from_icat(icat_data, icat_relations).json(
+            by_alias=True,
+        )
```

Comment:
We should not be passing ICAT relations to the `from_icat` method because it expects a list of the required related PaNOSC fields so this should be changed to:
```Python
        panosc_record = panosc_model.from_icat(icat_data, entity_relations).json(
            by_alias=True,
        )
```
You can get `entity_relations` before line 35 like this (I believe):
```Python
entity_relations = []
for filter_ in filters:
    if isinstance(filter_, SearchAPIIncludeFilter):
        entity_relations.extend(filter_.included_filters)
```

Reply:
Fixed in 4343a2c67cdba63a0b21e552b7d630eaf0b46cd6